### PR TITLE
"Gallows" mini-project/initial approach

### PR DESCRIPTION
**Pull request by VeroCodingProjects**

### CODE REVIEW

**Comment by keepcoding-tech** [comment]
Let's create a `img` folder for all of those images.

File: index.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        <script src="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/js/bootstrap.min.js"></script>
+    </head>
+    <body>
+        <form id="player1" action="javascript:startGame(this)">
+            <h1>GALLOWS</h1>
+            <input id="input" class="form-control form-control-lg" type="text" maxlength="20" required ="true" type="text" 
+                pattern="[a-zA-Z]+" oninvalid="setCustomValidity('Only letters are avaliable for this game.')">
+            <button class="btn btn-light">Start game</button>
+        </form>
+        <div id="player2">
```
Comment (keepcoding-tech) [inline index.html]: Let's find some better names for `player1` and `player2`

File: main.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+let player1 = document.getElementById("player1");
+let player2 = document.getElementById("player2");
+let wordToGuess;
+let mismatchedLetters = 0;
+let completeBoard = encrypted;
+let alphabetButtons = letters;
+let gallowsImages = images;
+let resetButton = restart;
+
+
```
Comment (keepcoding-tech) [inline main.js:28]: You don't need two empty line here

File: main.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+// for each letter in the word, a "_" will be displayed
+function completeArray(wordToGuess) {
+    for (let i = 0; i < wordToGuess.length; ++i) {
+        encryptedWord.push(" _ ");
+    }
+    displayEncryptedWord();
+}
+
+// depending on the number of lifes, completes the board and decides if the game is over or not
```
Comment (keepcoding-tech) [inline main.js]: Try to use maximum of 80 characters even for comments, it's good practice.

File: main.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        displayImages(mismatchedLetters);
+        if (mismatchedLetters == 8) {
+            mismatchedLetters = 9;
+            gameOver(mismatchedLetters);
+        }
+    }
+}
+
+// displayes the letters of the alphabet as buttons
+// adds functionality to the buttons
```
Comment (keepcoding-tech) [inline main.js:57]: displays* the letters of the alphabet as buttons and* adds functionality to the buttons

Please be careful with your spelling, the comments has to make sense to be useful.

File: main.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+function displayAlphabet() {
+    completeArray(wordToGuess);
```
Comment (keepcoding-tech) [inline main.js]: Why do you call the "completeArray" function here?

File: main.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    for (let i = 0; i < alphabet.length; ++i) {
+        const currentButton = document.createElement("button");
+        alphabetButtons.appendChild(currentButton);
+        currentButton.innerHTML = alphabet[i];
+        currentButton.id = alphabet[i];
+        currentButton.addEventListener("click", ()=> searchMatches(currentButton.id));
```
Comment (keepcoding-tech) [inline main.js]: You missed a space here

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: main.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    displayImages(mismatchedLetters);
+}
+
+// this is the main function of the game
+// for each clicked button:
+// → the encrypted word will be completed if it is the case
+// → hides the clicked button
+// → checks the status of the game
+// → displays the proper image
+function searchMatches(id) {
```
Comment (keepcoding-tech) [inline main.js]: Let's find a better name for this function, something like: "playerAction(id)"

File: main.js [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    switchDisplay();
+    encryptLetters(wordToGuess);
+    statusWord();
+    displayAlphabet();
+    displayImages(mismatchedLetters);
+}
+
+// this is the main function of the game
+// for each clicked button:
+// → the encrypted word will be completed if it is the case
```
Comment (keepcoding-tech) [inline main.js]: Use simple characters, so no arrows `->`

File: main.js [hunk 9]
```diff
@@ -0,0 +1,141 @@
+const alphabet = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"]
+const gallowImages = ["Hangman Images/1.jpg", "Hangman Images/2.jpg", "Hangman Images/3.jpg", "Hangman Images/4.jpg", "Hangman Images/5.jpg", "Hangman Images/6.jpg", "Hangman Images/7.jpg", "Hangman Images/8.jpg", "https://i.ebayimg.com/images/g/jQkAAOSwo-pjeWSL/s-l1600.jpg", "https://i.pinimg.com/236x/db/62/e0/db62e024c5cd86b8a14ec068b2318c85--captain-jack-sparrow-pirates-of-the-caribbean.jpg"];
+let encryptedWord = [];
+let player1 = document.getElementById("word-input");
+let player2 = document.getElementById("word-guessing");
```
Comment (keepcoding-tech) [inline main.js]: Let's find some better names for these 2 variables too.

File: main.js [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+        alphabetButtons.appendChild(currentButton);
+        currentButton.innerHTML = alphabet[i];
+        currentButton.id = alphabet[i];
+        currentButton.addEventListener("click", ()=> searchMatches(currentButton.id));
+        currentButton.className = "btn btn-light";
+    }
+}
+
+// for each letter in the word, one "-" will be displayed
+function statusWord() {
```
Comment (keepcoding-tech) [inline main.js]: What do you need this function for? Why do you need to display "-"?

File: main.js [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+        completeBoard.innerText += encryptedWord[i];
+    }   
+}
+
+// for each wrong letter, the man is closer to his death
+function displayImages(mismatchedLetters) {
+    gallowsImages.innerHTML = "";
+    let img = document.createElement("img");
+    img.src = gallowImages[mismatchedLetters];
+    gallowsImages.appendChild(img);
```
Comment (keepcoding-tech) [inline main.js]: Can't we only change the `src` style property instead of creating a new HTML element every time?

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: main.js (hunk 10 above)
Comment (VeroCodingProjects) [reply main.js]: The comment is not suggestive enough. The function rewrites the word at each button press if a letter is found. I changed the comment.

File: main.js (hunk 11 above)
Comment (VeroCodingProjects) [reply main.js]: right

File: main.js [hunk 12]
```diff
@@ -0,0 +1,138 @@
+const alphabet = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"]
+const gallowImages = ["img/1.jpg", "img/2.jpg", "img/3.jpg", "img/4.jpg", "img/5.jpg", "img/6.jpg", "img/7.jpg", "img/8.jpg", "https://t3.ftcdn.net/jpg/01/84/91/74/360_F_184917449_2UTmC5D609xMxaAyqGDzfNUt64CnOdFG.jpg", "https://fsb.zobj.net/crop.php?r=yDx9wB7yqfbxfzAok8T8bXK7tvVg7P24mhDNXvW32TKa1Vt9giET2X0_OQCBlI4gVTXFM3ShDs4xziVel1j0l1R1UY-xbs-wihPXPU4QW7FIGBvrBBGnkf-U-tn5oEylzKRJLNhcVeUBr3jisoCJOT5dOKKISBC2gAJznCClBiX2REVOTkQbbHtVBug"];
+let encryptedWord = [];
+let startPage = document.getElementById("word-input");
+let hangmanPage = document.getElementById("word-guessing");
```
Comment (keepcoding-tech) [inline main.js]: Let's use either "gallows" or "hangman"

File: main.js [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+let wordToGuess;
+let mismatchedLetters = 0;
+let completeBoard = encrypted;
+let alphabetButtons = letters;
+let gallowsImages = images;
+let resetButton = restart;
```
Comment (keepcoding-tech) [inline main.js]: Can you get rid of some global variables and instead send them as parameters where needed?

File: main.js [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+        currentButton.className = "btn btn-light";
+    }
+}
+
+// for each wrong letter, the man is closer to his death
+function displayImages(mismatchedLetters) {
+    images.src = gallowImages[mismatchedLetters];
+}
+
+// for each letter in the word, a "_" will be displayed
```
Comment (keepcoding-tech) [inline main.js]: Not a big deal, but a "[space]_[space]" will be displayed 😅

File: main.js [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+    completeBoard.innerHTML = "";
+    let matchingLetters = completeWord(letter);
+    markButton(letter);
+    statusWord();
+    countLifes(matchingLetters);
+    displayImages(mismatchedLetters);
+}
+
+// at each point in the game, the second player can reset the game
+resetButton.onclick = ()=> {
```
Comment (keepcoding-tech) [inline main.js]: You missed a space here, also, this isn't really a function, is a variable, so you can put it at the beginning of the file.

File: main.js [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+// this is the main function of the game
+// for each clicked button:
+// the encrypted word will be completed if it is the case
+// hides the clicked button
+// checks the status of the game
+// displays the proper image
+function playerGuess(letter) {
+    completeBoard.innerHTML = "";
```
Comment (keepcoding-tech) [inline main.js]: Why are you emptying this variable content if you will call the "statusWord" function anyway?

File: style.css [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+form {
+    text-align: center;
+}
+
+form input[type="text"] {
+    text-transform: uppercase;
+}
+
+body {
```
Comment (keepcoding-tech) [inline style.css]: you can combine this style with the one at line 1

File: style.css [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+    text-align: center;
+}
+
+h1 {
+    margin-top: 2cm;
+    margin-bottom: 2cm;
+}
+
+form {
```
Comment (keepcoding-tech) [inline style.css]: Same here, combine the styles, also, if you want a challenge, try to get rid of these styles but keep the same display. How do you think you can do that?

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: main.js [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+
+// displaying only the reset button and the last image if the game is over
+function hangmanStatus(mismatchedLetters) {
+    displayImages(mismatchedLetters);
+    encrypted.style.display = "none";
+    letters.style.display = "none";
+}
+
+// determine if the current character is a letter;
+function isLetter(char) {
```
Comment (keepcoding-tech) [inline main.js]: Can you get rid of this function? And no, I don't mean to move the code from this function where you need it, like, completely remove the code.

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

**Review by keepcoding-tech — APPROVED** [review]

**Pull request merged**